src/snippet/mod.rs

```diff
@@ -186,6 +186,29 @@ fn select_best_fragment_combination(fragments: &[FragmentCandidate], text: &str)
     }
 }
 
+/// Returns ranges that are collapsed into non-overlapped ranges.
+fn collapse_overlapped_ranges(ranges: &Vec<Range<usize>>) -> Vec<Range<usize>> {
```

Comment:
```suggestion
fn collapse_overlapped_ranges(ranges: &[Range<usize>]) -> Vec<Range<usize>> {
```

&Vec is almost never a good idea.

Reply:
Another solution would be for you to pass an iterator directly.

`impl Iterator<Item=Range<usize>>`